core/base/inc/TString.h

```diff
@@ -132,15 +138,27 @@ friend TString operator+(const TString &s1, const TString &s2);
 friend TString operator+(const TString &s,  const char *cs);
 friend TString operator+(const char *cs, const TString &s);
 friend TString operator+(const TString &s, char c);
-friend TString operator+(const TString &s, Long_t i);
-friend TString operator+(const TString &s, ULong_t i);
-friend TString operator+(const TString &s, Long64_t i);
-friend TString operator+(const TString &s, ULong64_t i);
 friend TString operator+(char c, const TString &s);
-friend TString operator+(Long_t i, const TString &s);
-friend TString operator+(ULong_t i, const TString &s);
-friend TString operator+(Long64_t i, const TString &s);
-friend TString operator+(ULong64_t i, const TString &s);
+
+template<class T>
+friend typename std::enable_if<ROOT::TypeTraits::IsSignedNumeral<T>::value,TString>::type
+operator+(TString s, T i);
```

Comment:
Why should this take a `TString` as first argument, not a `const TString&`? The latter is more performant (doesn't copy the string).

Reply:
For `operator+`, taking in the object by value is more performant because it will pass it through the output. It is like a `operator+=` but with copy elision. If it was `const TString&` we couldn't take advantage of this speed up.